code/src/main/orxAndroidNativeSupport.cpp

```diff
@@ -470,6 +731,36 @@ void android_main( android_app* state )
         orxAndroid_JNI_SetupThread(orxNULL);
     }
 
+    JNIEnv *env = Android_JNI_GetEnv();
+
+    jobject jActivity = state->activity->clazz;
+    sstAndroid.mActivity = env->NewGlobalRef(jActivity);
+
+    // Create Java-method access to the activity.getDeviceIds();
+    sstAndroid.midGetDeviceIds = env->GetMethodID(env->GetObjectClass(jActivity), "getDeviceIds", "()[I");
+
+    if(!sstAndroid.midGetDeviceIds) {
+        __android_log_print(ANDROID_LOG_WARN, "Orx", "Couldn't locate Java method getDeviceIds in the NativeActivity, check that they're named and typed correctly");
+    }
+
+    // Looper for Joystick lifecycle events
+    sstAndroid.looper = ALooper_prepare(ALOOPER_PREPARE_ALLOW_NON_CALLBACKS);
+    ALooper_addFd(sstAndroid.looper, sstAndroid.pipeJoystickEvent[0], LOOPER_ID_JOYSTICK_EVENT, ALOOPER_EVENT_INPUT, NULL, NULL);
+
+    // register jni methods manually becuase the automatic registration does not happens when Activity is not loading system library.
+    JNINativeMethod methodTable[]=
+      {
+          {"nativeOnPause", "()V", (void *)Java_nativeOnPause},

```

Comment:
nativeOnPause and nativeOnResume make no sense because sent as APP_CMD_PAUSE/APP_CMD_RESUME handled in handleCmd()  so I will remove them.